3rdparty/CEGUI/CMakeLists.txt

```diff
@@ -1,122 +1,75 @@
-    WORKING_DIRECTORY ${CEGUI_DEPS_BUILD_PATH}
-)
-
-execute_process(
-    COMMAND ${CMAKE_COMMAND} --build ${CEGUI_DEPS_BUILD_PATH} --config Release -j  ${IRR_PARALLEL_BUILD_JOBS}
-)
 
-message(STATUS "CEGUI 0.8.x dependencies built")
-
-message(STATUS "Installing CEGUI 0.8.x dependencies")
-execute_process(
-    COMMAND ${CMAKE_COMMAND} -E copy_directory
-    ${CEGUI_DEPS_BUILD_PATH}/dependencies
-    ${CEGUI_DEPS_INSTALL_PATH}
-)
-message(STATUS "Installing done")
-
-message(STATUS "Building CEGUI")
-
-execute_process(
-    COMMAND ${CMAKE_COMMAND} -E make_directory build
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-)
-
-# CEGUI needs at least 1 image codec to build although we won't use
-execute_process(
-    COMMAND ${CMAKE_COMMAND} ${CEGUI_PATH}
-        -DCMAKE_PREFIX_PATH=${CEGUI_PATH}/dependencies
-        -DCMAKE_INSTALL_PREFIX=${CEGUI_INSTALL_PATH}
-        -DCEGUI_INSTALL_PKGCONFIG=OFF
-        -DCEGUI_BUILD_APPLICATION_TEMPLATES=OFF
-        -DCEGUI_STRING_CLASS=2
-        -DCEGUI_HAS_FREETYPE=ON
-        -DCEGUI_HAS_PCRE_REGEX=ON
-        -DCEGUI_BUILD_PYTHON_MODULES=OFF
-        -DCEGUI_BUILD_LUA_MODULE=OFF
-        -DCEGUI_BUILD_XMLPARSER_XERCES=OFF
-        -DCEGUI_BUILD_XMLPARSER_TINYXML=OFF
-        -DCEGUI_BUILD_XMLPARSER_TINYXML2=OFF
-        -DCEGUI_BUILD_XMLPARSER_LIBXML2=OFF
-        -DCEGUI_BUILD_XMLPARSER_RAPIDXML=OFF
-        -DCEGUI_BUILD_XMLPARSER_EXPAT=ON
-        -DCEGUI_OPTION_DEFAULT_XMLPARSER=ExpatParser
-        -DCEGUI_BUILD_IMAGECODEC_STB=ON
-        -DCEGUI_BUILD_RENDERER_OPENGL=OFF
-        -DCEGUI_BUILD_RENDERER_OPENGL3=ON
-        -DCEGUI_SAMPLES_ENABLED=OFF
-        -DCEGUI_BUILD_STATIC_CONFIGURATION=ON
-        -DCEGUI_BUILD_STATIC_FACTORY_MODULE=ON
-        -DCEGUI_STATIC_XMLPARSER_MODULE=CEGUIExpatParser
-    WORKING_DIRECTORY ${CEGUI_BUILD_PATH}
-)
-
-execute_process(
-    COMMAND ${CMAKE_COMMAND}
-        --build ${CEGUI_BUILD_PATH}
-        --target install
-        --config Release
-        -j ${IRR_PARALLEL_BUILD_JOBS}
-)
-
-message(STATUS "CEGUI built")
-
-endif()
+# Disable unused CEGUI options
+set(CEGUI_INSTALL_PKGCONFIG OFF CACHE BOOL "")
+set(CEGUI_BUILD_APPLICATION_TEMPLATES OFF CACHE BOOL "")
+set(CEGUI_BUILD_PYTHON_MODULES OFF CACHE BOOL "")
+set(CEGUI_BUILD_LUA_MODULE OFF CACHE BOOL "")
+set(CEGUI_BUILD_XMLPARSER_XERCES OFF CACHE BOOL "")
+set(CEGUI_BUILD_XMLPARSER_TINYXML OFF CACHE BOOL "")
+set(CEGUI_BUILD_XMLPARSER_TINYXML2 OFF CACHE BOOL "")
+set(CEGUI_BUILD_XMLPARSER_LIBXML2 OFF CACHE BOOL "")
+set(CEGUI_BUILD_XMLPARSER_RAPIDXML OFF CACHE BOOL "")
+set(CEGUI_BUILD_RENDERER_OPENGL OFF CACHE BOOL "")
+set(CEGUI_SAMPLES_ENABLED OFF CACHE BOOL "")
+
+# Enable required CEGUI options
+set(CEGUI_STRING_CLASS "2" CACHE STRING "")
+set(CEGUI_HAS_FREETYPE ON CACHE BOOL "")
+set(CEGUI_HAS_PCRE_REGEX ON CACHE BOOL "")
+set(CEGUI_BUILD_XMLPARSER_EXPAT ON CACHE BOOL "")
+set(CEGUI_BUILD_IMAGECODEC_STB ON CACHE BOOL "") # needs at least 1 image codec module
```

Comment:
eventually after hooking in irrbaw image loaders, this will disappear?

Reply:
No, it needs to compile because the code must access to the image codec interface.
Unless I patched CEGUI to have a null image codec or embed irrbaw image loader code into CEGUI (not 100% certain if that will be accepted into original CEGUI repo).